Microsoft.Toolkit.Uwp.Services/Services/LinkedIn/LinkedInDataProvider.cs

```diff
@@ -0,0 +1,317 @@
+using Windows.Security.Credentials;
+using Windows.Storage;
+using Windows.Web.Http;
+
+namespace Microsoft.Toolkit.Uwp.Services.LinkedIn
+{
+    /// <summary>
+    /// Data Provider for connecting to LinkedIn service.
+    /// </summary>
+    public class LinkedInDataProvider
+    {
+        private const string OAuthBaseUrl = "https://www.linkedin.com/uas/oauth2/";
+        private const string BaseUrl = "https://api.linkedin.com/v1";
+
+        /// <summary>
+        /// Tokens for service.
+        /// </summary>
+        private readonly LinkedInOAuthTokens tokens;
+
+        /// <summary>
+        /// Password vault used to store access tokens
+        /// </summary>
+        private readonly PasswordVault vault;
+
+        /// <summary>
+        /// Gets or sets logged in user information.
+        /// </summary>
+        public string Username { get; set; }
+
+        /// <summary>
+        /// Gets a value indicating whether the provider is already logged in
+        /// </summary>
+        public bool LoggedIn { get; private set; }
+
+        /// <summary>
+        /// Gets or sets requiredPermissions.
+        /// </summary>
+        public LinkedInPermissions RequiredPermissions { get; set; }
+
+        /// <summary>
+        /// Initializes a new instance of the <see cref="LinkedInDataProvider"/> class.
+        /// Constructor.
+        /// </summary>
+        /// <param name="tokens">OAuth tokens for request.</param>
+        /// <param name="requiredPermissions">Required permissions for the session.</param>
+        public LinkedInDataProvider(LinkedInOAuthTokens tokens, LinkedInPermissions requiredPermissions)
+        {
+            this.tokens = tokens;
+            this.RequiredPermissions = requiredPermissions;
+
+            vault = new PasswordVault();
+        }
+
+        private PasswordCredential PasswordCredential
+        {
+            get
+            {
+                // Password vault remains when app is uninstalled so checking the local settings value
+                if (ApplicationData.Current.LocalSettings.Values[LinkedInConstants.STORAGEKEYUSER] == null)
+                {
+                    return null;
+                }
+
+                var passwordCredentials = vault.RetrieveAll();
+                var temp = passwordCredentials.FirstOrDefault(c => c.Resource == LinkedInConstants.STORAGEKEYACCESSTOKEN);
+
+                if (temp == null)
+                {
+                    return null;
+                }
+
+                return vault.Retrieve(temp.Resource, temp.UserName);
+            }
+        }
+
+        /// <summary>
+        /// Log user in to LinkedIn.
+        /// </summary>
+        /// <param name="force">Force login.</param>
+        /// <returns>Boolean indicating login success.</returns>
+        public async Task<bool> LoginAsync(bool force = false)
+        {
+            var linkedInCredentials = PasswordCredential;
+            if (linkedInCredentials != null && !force)
```

Comment:
we should discuss / consider for a subsequent update as having this level of consistency is sometimes out of our control where we are using an underlying provider.